lib/ddtrace/version.rb

```diff
@@ -10,5 +10,9 @@ module VERSION
     # Support for Ruby < 2.1 is currently deprecated in the tracer.
     # Support will be dropped in the near future.
     MINIMUM_RUBY_VERSION = '2.0.0'.freeze
+
+    # Ruby 3.2 is not supported, given kwargs transition and disappearance of ruby2_keywords
+    # Support will be extended in the near future
+    MAXIMUM_RUBY_VERSION = '3.2.a'.freeze
```

Comment:
`< 3.2` would match `3.2.0-preview1`, mistakenly so (we want it excluded). Thats how `rubygems` [encodes the minimal version including `prerelease?` ones](https://github.com/rubygems/rubygems/blob/c5c4c62547444330c170ab227a8798bc503c6db5/lib/rubygems/version.rb#L26-L29) (interestingly enough, ["no version requirement" is in fact `Gem::Version.new('0.a')`](https://github.com/rubygems/rubygems/blob/c5c4c62547444330c170ab227a8798bc503c6db5/lib/rubygems/requirement.rb#L40)).

Trivia: interestingly enough, because of the `-p648`:

```
irb> Gem::Version.new('2.0.0p648').prerelease?
=> true
irb> Gem::Version.new('2.0.0p648') < Gem::Version.new('2.0')
=> true
```

Which, as you see, had the potency to cause shenanigans in comparisons if you weren't careful ;)

Reply:
Thanks! I totally did now know that.

TBH, I don't quite mind if people use it with preview versions -- it's a Ruby preview version, so you know going in that "things can break" -- so I'd still go with the 3.2, but I think 3.2.a is very reasonable as well so I'll let you make the final decision :)

Reply:
Ultimately I went for `3.2`, as I thought they might eventually delay the drop if enough gems have not made the move. This way this can be easily tested for that before the final 3.2 release, as well as not blocking early testing of non-kwargs dependent integrations on the preview.